### ISSUE TITLE
Questions about image order and prior_* columns in database

### ISSUE THREAD

**Issue by Beniko95J** [colmap/colmap#642]
1. I found that the order of image ids in the database is sometimes the same as the order of their names(which are their timestamps), while sometimes not. like this case is not well ordered:
![1](https://user-images.githubusercontent.com/30772597/60644221-5fa9cc80-9e70-11e9-8344-214318a44ff7.PNG)
this is well ordered:
![2](https://user-images.githubusercontent.com/30772597/60644251-77815080-9e70-11e9-969e-26c2e75dc604.PNG)
I hope the order of image ids in the database is always the same order as their names (which contain their timestamps). Is there any way to do this?

2. I am going to use relative poses estimated by visual-inertial odometry as initial image poses for better image registration. I found that there have been columns like prior_tx, prior_ty and etc. in the images table in the database, but it seems like that colmap never uses them in its reconstruction pipeline, isn't it? If not, I will need to be careful about change these values in the database without changing the current behavior of colmap. I would like to confirm it with you.

Thanks!

**Comment by silence401** [comment]
I also found the database sometimes is not ordered and maybe this link can solve the second problem for you ? [https://github.com/colmap/colmap/issues/559#issuecomment-457926309](url)

**Comment by Beniko95J** [comment]
@silence401 

This link seems to be invalid. Could you please send it again?

**Comment by silence401** [comment]
https://github.com/colmap/colmap/issues/559#issuecomment-457926309

**Comment by Beniko95J** [comment]
@silence401 

Thanks! This comment is very helpful. By the way, are the prior poses needed for **spatial matcher** the projection from world to the camera coordinate system or from camera to world?

**Comment by silence401** [comment]
> @silence401
> 
> Thanks! This comment is very helpful. By the way, are the prior poses needed for **spatial matcher** the projection from world to the camera coordinate system or from camera to world?

I am not very clear about it, you can see the document about colmap ,there is a decription about spatial matcher

**Comment by ahojnnes** [comment]
1. The image_id in the database cannot be assumed to be ordered. 
2. The prior_q* values are not used anywhere in the pipeline. Simple IMU rotation/translation constraints could be easily added to the bundle adjuster though.
3. The prior_t* values are used in the spatial matcher but this more intended for GPS like data to determine which images to match.

**Comment by silence401** [comment]
>     1. The image_id in the database cannot be assumed to be ordered.
> 
>     2. The prior_q* values are not used anywhere in the pipeline. Simple IMU rotation/translation constraints could be easily added to the bundle adjuster though.
> 
>     3. The prior_t* values are used in the spatial matcher but this more intended for GPS like data to determine which images to match.
Does the reconstruct sparse/dense model from known camera poses pipeline use the prior_q*  and prior_t* values ? And where to use them.

**Comment by ahojnnes** [comment]
No, it doesn’t. It uses the images/cameras.txt/.bin. Please follow the instructions in the FAQ for this.

**Comment by silence401** [comment]
> images

Thanks for your reply,I just use the GPS information as the camera pose and write it in the images.txt.But the reconstruct result absolutely wrong.Does the GPS information need some translations to be used in the pipeline ? Thanks.

**Comment by silence401** [comment]
> No, it doesn’t. It uses the images/cameras.txt/.bin. Please follow the instructions in the FAQ for this.
Thanks for your reply,I just use the GPS information as the camera pose and write it in the images.txt.But the reconstruct result absolutely wrong.Does the GPS information need some translations to be used in the pipeline ? Thanks.

**Comment by YasmineCupcake** [comment]
I wish this would be mentioned!! I spent hours on this to realize at the end that the pictures aren't ordered and that images_id isn't the real order of the images i put as input, a bit confusing ngl

**Comment by YasmineCupcake** [comment]
never mind sir it loaded them all in order, IM SO GLAD UGH 

<img width="282" height="606" alt="Image" src="https://github.com/user-attachments/assets/5524620a-3cbf-4217-af58-54d423a589c9" />

**Comment by ChEnYoNgB** [comment]
> never mind sir it loaded them all in order, IM SO GLAD UGH
> 
> <img alt="Image" width="282" height="606" src="https://private-user-images.githubusercontent.com/114703736/524484527-5524620a-3cbf-4217-af58-54d423a589c9.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Njk1NzQ5NTQsIm5iZiI6MTc2OTU3NDY1NCwicGF0aCI6Ii8xMTQ3MDM3MzYvNTI0NDg0NTI3LTU1MjQ2MjBhLTNjYmYtNDIxNy1hZjU4LTU0ZDQyM2E1ODljOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwMTI4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDEyOFQwNDMwNTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04NTdkNmI5OWYwMzVhMjVkM2UzOTExMTZlZmMxNTU1ZTkyYTcwMDFkZWNiNWYxNWJmZTY2ZWFkY2M4MTU1Y2ViJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.loKdMSbs1XLiBxD4w00bnuaKqZKVtex76ZHfHG4G7mE">

Could you please tell me how you managed to achieve this?

**Comment by YasmineCupcake** [comment]
> > never mind sir it loaded them all in order, IM SO GLAD UGH
> > 
> > <img alt="Image" width="282" height="606" src="https://private-user-images.githubusercontent.com/114703736/524484527-5524620a-3cbf-4217-af58-54d423a589c9.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Njk1NzQ5NTQsIm5iZiI6MTc2OTU3NDY1NCwicGF0aCI6Ii8xMTQ3MDM3MzYvNTI0NDg0NTI3LTU1MjQ2MjBhLTNjYmYtNDIxNy1hZjU4LTU0ZDQyM2E1ODljOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwMTI4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDEyOFQwNDMwNTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04NTdkNmI5OWYwMzVhMjVkM2UzOTExMTZlZmMxNTU1ZTkyYTcwMDFkZWNiNWYxNWJmZTY2ZWFkY2M4MTU1Y2ViJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.loKdMSbs1XLiBxD4w00bnuaKqZKVtex76ZHfHG4G7mE">
> 
> Could you please tell me how you managed to achieve this?

Of course, when you finish the colmap process you will get a .db file, I just loaded it into python with simple instructions to list the image names and i found that they were loaded in the correct order

**Comment by ahojnnes** [comment]
Closing as the maintainer answered all three questions: image_id ordering cannot be assumed, prior_q* is unused, and prior_t* is only used in spatial matcher.
